contracts/src/Utilities/ValidatorSetSig.sol

```diff
@@ -0,0 +1,137 @@
+ * to deploy a `Ownable` target contract that is owned by an instance of this contract and adding the
+ * `onlyOwner` modifier to the functions that should be governed.
+ *
+ * @custom:security-contact https://github.com/ava-labs/teleporter/blob/main/SECURITY.md
+ */
+contract ValidatorSetSig is ReentrancyGuard {
+    /**
+     * @notice A blockchain ID whose validators need to sign the message for it to be considered valid.
+     */
+    bytes32 public immutable validatorBlockchainID;
+
+    /**
+     * @dev Tracks nonces for messages sent to a specific contract address to provide replay protection.
+     * The target contract address is used as the key instead of keeping a global nonce to prevent
+     * one party from consuming a global nonce that another was intending to use
+     * in case a single instance of ValidatorSetSig contract is used to manage multiple downstream contracts.
+     */
+    mapping(address targetContractAddress => uint256 nonce) public nonces;
+
+    /**
+     * @notice Address that the off-chain Warp message sets as the "source" address.
+     * @dev The address is not owned by any EOA or smart contract account, so it
+     * cannot possibly be the source address of any other Warp message emitted by the VM.
+     */
+    address public constant VALIDATORS_SOURCE_ADDRESS = address(0);
+
+    /**
+     * @notice The blockchain ID of the chain the contract is deployed on.
+     */
+    bytes32 public immutable blockchainID;
+
+    /**
+     * @notice Warp precompile used for sending and receiving Warp messages.
+     */
+    IWarpMessenger public constant WARP_MESSENGER =
+        IWarpMessenger(0x0200000000000000000000000000000000000005);
+
+    /**
+     * @notice Emited when the payload is successfully delivered to the target contract.
+     */
+    event Delivered(address indexed targetContractAddress, uint256 indexed nonce);
+
+    constructor(bytes32 validatorBlockchainID_) {
+        validatorBlockchainID = validatorBlockchainID_;
+        blockchainID = WARP_MESSENGER.getBlockchainID();
+    }
+
+    function executeCall(uint32 messageIndex) external nonReentrant {
+        // Get the WarpMessage from the WarpMessenger pre-compile and verify that it is valid
+        (WarpMessage memory message, bool valid) =
+            WARP_MESSENGER.getVerifiedWarpMessage(messageIndex);
+        require(valid, "ValidatorSetSig: invalid warp message");
+
+        // Ensure that the sourceChainID of the validator set that signed the message is the authorized one
+        require(
+            message.sourceChainID == validatorBlockchainID, "ValidatorSetSig: invalid sourceChainID"
+        );
+        // Ensure that the originSenderAddress is the zero address to prevent on-chain messages from being forwarded
+        require(
+            message.originSenderAddress == VALIDATORS_SOURCE_ADDRESS,
+            "ValidatorSetSig: non-zero originSenderAddress"
```

Comment:
```suggestion
            "ValidatorSetSig: non-zero origin sender address"
```
we generally havent' used the variable names in revert messages

Reply:
I think it makes sense for very user facing transactions but anyone seeing this is manually constructing a off-chain warp message and seeing the variable name gives them something to search the source code for. It's even more pronounced in later require message of `invalid validatorSetSigAddress`. IMO `invalid validator set sig address` is less clear and useful. And explaining of what the field is in the revert message would be even less clear? `warp message used is addressed to a different validator set sig contract address`

Reply:
that's fair in regards to easier searching or relating to the variable, sgtm.